sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -296,6 +325,9 @@ export class EventHubReceiver extends LinkEntity {
       const receiverError = context.receiver && context.receiver.error;
       const receiver = this._receiver || context.receiver!;
       if (receiverError) {
+        if (this._aborter) {
+          this._aborter.removeEventListener("abort", this._onAbort);
+        }
```

Comment:
If the receiver is being reconnected (because `detached()` was called for a previous error) or if decide to call `detached()` for the current error, then we shouldnt remove the listener. Because the reconnected receiver should still respect the cancellation token.

So please update the changes in both `_onAmqpClose` and `_onSessionClose` appropriately